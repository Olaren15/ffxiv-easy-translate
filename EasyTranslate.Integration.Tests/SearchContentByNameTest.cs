using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Threading.Tasks;
using EasyTranslate.Domain.Entities;
using EasyTranslate.UseCase;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace EasyTranslate.Integration.Tests;

// Assertions are done with snapshots
[SuppressMessage("Blocker Code Smell", "S2699:Tests should include assertions")]
public class SearchContentByNameTest
{
    private readonly SearchContentByNameUseCase _searchContentByNameUseCase;

    public SearchContentByNameTest()
    {
        IServiceProvider serviceProvider = ServiceProviderBuilder.Build();
        _searchContentByNameUseCase = serviceProvider.GetService<SearchContentByNameUseCase>()!;
    }

    [Fact]
    public async Task Item_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Popoto",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Action_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "cure",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }


    [Fact]
    public async Task Achievement_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For the hoard",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task BNpcname_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "bhoot",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ClassJob_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute("summoner", Language.English);

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Companion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "lesser panda",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ContentFinderCondition_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "sastasha",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task CraftAction_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Groundwork",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Emote_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pet",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ENpcResident_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "G'raha Tia",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task EventItemLite_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Ulan's note",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Fate_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Drink Me",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Leve_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Practical Command",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Mount_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Demi-Ozma",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Orchestrion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pa-paya",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task PlaceName_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Limsa Lominsa",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Quest_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For Want Of a Memory",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Status_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Vulnerability Up",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Title_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Goddess of Magic",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Trait_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Maim and Mend",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Weather_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Fair Skies",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }
}
