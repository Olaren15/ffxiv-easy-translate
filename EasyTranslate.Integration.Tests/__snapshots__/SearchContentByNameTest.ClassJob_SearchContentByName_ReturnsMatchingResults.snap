[
  {
    "Type": "ClassJob",
    "IconId": 62127,
    "EnglishName": "summoner",
    "FrenchName": "invocateur",
    "GermanName": "Beschwörer",
    "JapaneseName": "召喚士"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "Joi summoner",
    "FrenchName": "kuchiyose-ninja isolationniste",
    "GermanName": "geisterflüsternd[a] Ninja",
    "JapaneseName": "攘夷派の口寄忍者"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "dark summoner",
    "FrenchName": "invocateur ténébreux",
    "GermanName": "unbekannt[a] Beschwörer",
    "JapaneseName": "黒い召喚士"
  },
  {
    "Type": "Title",
    "IconId": null,
    "EnglishName": "Makai Summoner",
    "FrenchName": "Invocateur Makai / Invocatrice Makai",
    "GermanName": "Makai-Beschwörer / Makai-Beschwörerin",
    "JapaneseName": "Makai Summoner"
  },
  {
    "Type": "Achievement",
    "IconId": 26010,
    "EnglishName": "Soul Summoner I",
    "FrenchName": "Modèle d'excellence de l'invocateur",
    "GermanName": "Seelenbeschwörer",
    "JapaneseName": "正道の召喚士"
  },
  {
    "Type": "Item",
    "IconId": 41180,
    "EnglishName": "Summoner's Horn",
    "FrenchName": "Corne de l'invocateur",
    "GermanName": "Beschwörer-Horn",
    "JapaneseName": "サマナーホーン"
  },
  {
    "Type": "Achievement",
    "IconId": 26010,
    "EnglishName": "Soul Summoner II",
    "FrenchName": "Invocateur consacré",
    "GermanName": "Seelenbeschwörer II",
    "JapaneseName": "龍神の召喚士"
  },
  {
    "Type": "Achievement",
    "IconId": 26010,
    "EnglishName": "Soul Summoner IV",
    "FrenchName": "Maître invocateur",
    "GermanName": "Seelenbeschwörer IV",
    "JapaneseName": "先達の召喚士"
  },
  {
    "Type": "Achievement",
    "IconId": 26010,
    "EnglishName": "Soul Summoner III",
    "FrenchName": "Invocateur illustre",
    "GermanName": "Seelenbeschwörer III",
    "JapaneseName": "最強の召喚士"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "spectral summoner",
    "FrenchName": "invocatrice spectrale",
    "GermanName": "Phantom-Beschwörer",
    "JapaneseName": "幻光の召喚士"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "ebon-clad summoner",
    "FrenchName": "invocateur vêtu de noir",
    "GermanName": "schwarz gekleidet[a] Beschwörer",
    "JapaneseName": "黒法衣の召喚士"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "High Summoner Sari",
    "FrenchName": "grand invocateur Sari",
    "GermanName": "[t] groß[a] Sari",
    "JapaneseName": "大召喚士サリ"
  },
  {
    "Type": "Item",
    "IconId": 43632,
    "EnglishName": "Summoner's Doublet",
    "FrenchName": "Doublet de l'invocateur",
    "GermanName": "Beschwörer-Wams",
    "JapaneseName": "サマナーダブレット"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "off-center summoner",
    "FrenchName": "invocateur mystérieux",
    "GermanName": "geheimnisvoll[a] Beschwörer",
    "JapaneseName": "謎めいた召喚士"
  },
  {
    "Type": "Item",
    "IconId": 26010,
    "EnglishName": "Soul of the Summoner",
    "FrenchName": "Cristal d'invocateur",
    "GermanName": "Beschwörer-Kristall",
    "JapaneseName": "召喚士の証"
  },
  {
    "Type": "Item",
    "IconId": 44212,
    "EnglishName": "Summoner's Ringbands",
    "FrenchName": "Brassières de l'invocateur",
    "GermanName": "Beschwörer-Ringbänder",
    "JapaneseName": "サマナーバンド"
  },
  {
    "Type": "Item",
    "IconId": 45369,
    "EnglishName": "Summoner's Waistclout",
    "FrenchName": "Demi-guêtres de l'invocateur",
    "GermanName": "Beschwörer-Lendentuch",
    "JapaneseName": "サマナーウェストクラウト"
  },
  {
    "Type": "Item",
    "IconId": 46369,
    "EnglishName": "Summoner's Thighboots",
    "FrenchName": "Pigaches de l'invocateur",
    "GermanName": "Beschwörer-Kniestiefel",
    "JapaneseName": "サマナーサイブーツ"
  },
  {
    "Type": "Item",
    "IconId": 43739,
    "EnglishName": "High Summoner's Dress",
    "FrenchName": "Robe de grande invocatrice",
    "GermanName": "Robe des Hohen Mediums",
    "JapaneseName": "ハイサマナー・ドレス"
  },
  {
    "Type": "Item",
    "IconId": 46665,
    "EnglishName": "High Summoner's Boots",
    "FrenchName": "Bottes de grande invocatrice",
    "GermanName": "Stiefel des Hohen Mediums",
    "JapaneseName": "ハイサマナー・ブーツ"
  },
  {
    "Type": "Item",
    "IconId": 26625,
    "EnglishName": "Summoner Framer's Kit",
    "FrenchName": "L'Art du portrait : invocateur",
    "GermanName": "Portraitmaterial (Beschwörer)",
    "JapaneseName": "ポートレート教材:召喚士"
  },
  {
    "Type": "Item",
    "IconId": 43739,
    "EnglishName": "High Summoner's Attire",
    "FrenchName": "Tenue de grande invocatrice",
    "GermanName": "Kleidungsset des Hohen Mediums",
    "JapaneseName": "ハイサマナーアタイア"
  },
  {
    "Type": "Quest",
    "IconId": 71221,
    "EnglishName": "Shadowing the Summoner",
    "FrenchName": "À la poursuite d'une ombre",
    "GermanName": "Auf Tristans Spuren",
    "JapaneseName": "黒き召喚士の影"
  },
  {
    "Type": "Item",
    "IconId": 44490,
    "EnglishName": "High Summoner's Armlets",
    "FrenchName": "Manches cloches de grande invocatrice",
    "GermanName": "Armreife des Hohen Mediums",
    "JapaneseName": "ハイサマナー・アームレット"
  },
  {
    "Type": "Item",
    "IconId": 53503,
    "EnglishName": "Summoner Display Crystal",
    "FrenchName": "Trophée en cristal d'invocateur",
    "GermanName": "Beschwörer-Tischkristall",
    "JapaneseName": "クリスタルトロフィー:召喚士"
  },
  {
    "Type": "Item",
    "IconId": 26509,
    "EnglishName": "Summoner's Attire Coffer (IL 90)",
    "FrenchName": "Coffre d'équipement de l'invocateur [NvObj 90]",
    "GermanName": "Beschwörer-Ausrüstungsset (G.-St. 90)",
    "JapaneseName": "サマナーチェスト【ILv90】"
  },
  {
    "Type": "Item",
    "IconId": 26410,
    "EnglishName": "Tales of Adventure: One Summoner's Journey I",
    "FrenchName": "“Contes d'aventures : Périple d'un invocateur I”",
    "GermanName": "Abenteuergeschichten: Reise eines Beschwörers I",
    "JapaneseName": "召喚士の冒険録:蒼天編"
  },
  {
    "Type": "Item",
    "IconId": 26410,
    "EnglishName": "Tales of Adventure: One Summoner's Journey II",
    "FrenchName": "“Contes d'aventures : Périple d'un invocateur II”",
    "GermanName": "Abenteuergeschichten: Reise eines Beschwörers II",
    "JapaneseName": "召喚士の冒険録:紅蓮編"
  },
  {
    "Type": "Item",
    "IconId": 26410,
    "EnglishName": "Tales of Adventure: One Summoner's Journey IV",
    "FrenchName": "“Contes d'aventures : Périple d'un invocateur IV”",
    "GermanName": "Abenteuergeschichten: Reise eines Beschwörers IV",
    "JapaneseName": "召喚士の冒険録:暁月編"
  },
  {
    "Type": "Item",
    "IconId": 26410,
    "EnglishName": "Tales of Adventure: One Summoner's Journey III",
    "FrenchName": "“Contes d'aventures : Périple d'un invocateur III”",
    "GermanName": "Abenteuergeschichten: Reise eines Beschwörers III",
    "JapaneseName": "召喚士の冒険録:漆黒編"
  }
]
